dsc/workflows/base/__init__.py

```diff
@@ -300,33 +319,42 @@ def process_sqs_queue(self) -> list[tuple[str, dict]]:
                 item_identifier, message_body = sqs_client.process_result_message(
                     sqs_message
                 )
-                items.append((item_identifier, message_body))
-                logger.debug(item_identifier, message_body)
-                self.report_data.append(f"{item_identifier}: {message_body}")
+                items.append(
+                    {"item_identifier": item_identifier, "message_body": message_body}
+                )
+                self.workflow_events.processed_items.append(
+                    {
+                        "item_identifier": item_identifier,
+                        "doi": message_body["ItemHandle"],
+                    }
+                )
```

Comment:
These two objects look _very_ similar.  This also goes back to an interesting discussion I recall having with @ehanson8 about passing data vs reporting.

For the `WorkflowEvents`, do we _want_ to accumlate objects like this?  Or might we want more human-friendly strings?

**Argument for objects**: this gives the report the ability to change what it retrieves from the object later, and how it displays that.

**Argument for strings**: the workflow is deciding how to "spin" or describe the event, and it's a static string that it provides.

I don't think there is a "correct" answer, just an opinionated one.  But I do think answering that question has bearing on how these two objects look very similar.

Reply:
For the reasons shared above, I think I prefer **objects**. It seems like it would be easier to either create a custom report class or update a template vs. changing a string in `WorkflowEvents` (and in `Workflow` method calls where `WorkflowEvents` is populated) whenever a rewording is needed. Similarly, adding a data point to `WorkflowEvents` if something is missing and then updating the report template feels more flexible...

What do you think, @ehanson8 and @ghukill ? 🤔

Reply:
@jonavellecuerdo - objects make sense to me!  I guess my followup regarding their _near_ similarity: why not use the same object for both then?  It looks like the reporting  object just  is a _subset_ of the SQS message body, but if you pass the whole thing, then reporting has more options on what to selectively report on.